### ISSUE TITLE
LMTP server offers HELO, doesn't provide correct LHLO response

### ISSUE THREAD

**Issue by tlhackque** [aio-libs/aiosmtpd#113]
As initailly reported here: https://gitlab.com/mailman/mailman/issues/348

Telnet to the LMTP server, and type "HELP".

It responds with 
250 Supported commands: EHLO HELO MAIL RCPT DATA RSET NOOP QUIT VRFY

This isn't correct.  LMTP uses LHLO; the EHLO and HELO should not be offered.

LHLO foo

Responds with
250 foo.example.net

RFC2033 says the implementation MUST support PIPELINING and ENHANCEDSTATUSCODES, and SHOULD support 8BITMIME

Thus, LHLO should be responding with the ESMTP keywords for these extensions.

8BITMIME (RFC 1652)

ENHANCEDSTATUSCODES (RFC 2034)

PIPELINING (RFC 1854)

Since it also responds to HELP HELP should also be listed (RFC 1869)

Thus, the dialog should look like:
````
S: 220 foo.example.net GNU Mailman LMTP ...
C: LHLO foo
S: 250-foo.example.net
S: 250-HELP
S: 250-ENHANCEDSTATUSCODES
S: 250-8BITMIME
S: 250 PIPELINING
````

(The order doesn't matter.)

And, obviously, 'HELP' should respond with the correct list of supported commands - including LHLO.

**Comment by Varbin** [comment]
See also #69.

**Comment by warsaw** [comment]
@Varbin Yes, we should fix this as part of #69 .  I think the decorator syntax could work, but it needs a little more tweaking to support distinguishing between EHLO and LHLO.
